src/components/accessories/admin/users/editUser/EditUser.tsx

```diff
@@ -0,0 +1,64 @@
+import { CircularProgress } from "@mui/material";
+import { useAppDispatch, useAppSelector } from "libraries/hooks/redux";
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router";
+import { Navigate } from "react-router-dom";
+
+import { PATHS } from "../../../../../consts";
+import { UserDTO } from "../../../../../generated";
+import { getUserGroups } from "../../../../../state/usergroups";
+import { getUsers, updateUser, updateUserReset } from "../../../../../state/users";
+import { EditUserForm } from "./EditUserForm";
+
+export const EditUser = () => {
+  const dispatch = useAppDispatch();
+  const { id } = useParams();
+  const [user, setUser] = useState<UserDTO | null>();
+  const [userNotFound, setUserNotFound] = useState<boolean>(false);
+  const { isLoading, hasSucceeded, hasFailed, error } = useAppSelector(
+    (state) => state.users.update
+  );
```

Comment:
You're not using these states vars in this component, but you're just passing to the child component. I think you should better fetch them in the child component directly from the store.

Reply:
That way I try to keep separated:
1. data concerns: querying from API, managing redux store
2. ui and local state concerns: formik, display of user groups, etc.

It is intended to provide a clearer and more readable code, but if it makes you confuse, then it totally fails its purpose and should be refactored, wdyt?

Reply:
Well, let's just say we've been doing it ever since, and I don't think it's a problem. If you read a state slice just to pass it to a child component (which I consider as a kind of `props drilling`), I think you could better read the state directly within the child component.

Reply:
useFormik does not accept new initalProps, only the ones available on mount